content/posts/2020/2020-02-19-dask-clusters-overview.md

```diff
@@ -0,0 +1,337 @@
+
+```python
+from dask.distributed import Client
+from dask_cloudprovider import FargateCluster
+
+cluster = FargateCluster()
+client = Client(cluster)
+```
+
+This cluster manager uses your [AWS credentials](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html#cli-quick-configuration) to authenticate and request AWS resources on Fargate, and then connects your local session to the Dask cluster running on the cloud.
+
+There are even higher level services such as [AWS Lambda](https://aws.amazon.com/lambda/) or [Google Cloud Functions](https://cloud.google.com/functions) which allow you to execute code on demand and you are billed for the execution time of the code. These are referred to as "serverless" services as the servers are totally abstracted away. This would be perfect for out Dask cluster as you could submit the scheduler and workers as the code to run. **However** when running these cloud functions it is not possible to get a network connection between them as they do not have routable IP addresses, so there is no way to set up a Dask cluster made of these executing functions. Maybe one day!
+
+### Dask Gateway
+
+[Dask Gateway](https://gateway.dask.org/) is a central service for managing Dask clusters. It provides a secure API which multiple users can communicate with to request Dask servers. It can spawn Dask clusters on Kubernetes, Yarn or batch systems.
+
+This tool is targeted at IT administrators who want to enable their users to create Dask clusters, but want to maintain some centralized control instead of each user creating their own thing. This can also be useful for tracking Dask usage and setting per user limits.
+
+```python
+from dask.distributed import Client
+from dask_gateway import GatewayCluster
+
+cluster = GatewayCluster()
+client = Client(cluster)
+```
+
+For each user the commands for creating and using a gateway cluster are the same. It is down to the administrator to setup and manage the gateway server and configure [authentication via kerberos or Jupyter Hub](https://gateway.dask.org/authentication.html#). They should also provide [configuration](https://gateway.dask.org/configuration-user.html) to their users so that Dask Gateway knows how to connect to the gateway server. In a large organisation or institution the IT department also likely provisions the machines that staff are using, and so should be able to drop configuration files onto users computers.
+
+### Local CUDA Cluster
+
+The last cluster manager I'm going to cover is `LocalCUDACluster` from the [Dask CUDA](https://github.com/rapidsai/dask-cuda) package.
+
+This is slightly different than the other cluster managers in that it is constructing a Dask cluster which is specifically optimised for a single piece of hardware. In this case it is targeting machines with multiple GPUs like an [NVIDIA DGX-2](https://www.nvidia.com/en-gb/data-center/dgx-2/).
```

Comment:
Maybe here we should be more generic? Specifically pointing to a DGX-2 may give the impression users need a big machine, but they could be using that on any machine with at least one NVIDIA GPU, even a laptop.

Reply:
Good point!

Reply:
```suggestion
This is slightly different than the other cluster managers in that it is constructing a Dask cluster which is specifically optimised for a single piece of hardware. In this case it is targeting machines with GPUs ranging from your laptop with an onboard NVIDIA GPU to an [NVIDIA DGX-2](https://www.nvidia.com/en-gb/data-center/dgx-2/) with multiple GPUs running in your datacentre.
```